lib/filterTokens.js

```diff
@@ -53,8 +53,11 @@ async function filterTokenObject(tokens, filter, options) {
     if (!isPlainObject(token)) {
       return acc;
     } else {
-      const tokenValue = options.usesDtcg ? token.$value : token.value;
-      if (typeof tokenValue === 'undefined') {
+      const valueKey = options.usesDtcg ? '$value' : 'value';
+      const tokenValue = token?.[valueKey];
+      const hasNestedValue = tokenValue && Object.hasOwn(tokenValue, valueKey);
```

Comment:
Given how often this particular code is repeated, I feel like we should abstract it to a utility function `isToken(token, usesDtcg)`